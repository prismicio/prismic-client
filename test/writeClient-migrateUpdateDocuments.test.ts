import {
	testMigrationImageFieldPatching,
	testMigrationLinkFieldPatching,
	testMigrationSimpleFieldPatching,
} from "./__testutils__/testMigrationFieldPatching"

testMigrationSimpleFieldPatching("does not patch simple fields", {
	embed: ({ ctx }) => ctx.mock.value.embed({ state: "filled" }),
	date: ({ ctx }) => ctx.mock.value.date({ state: "filled" }),
	timestamp: ({ ctx }) => ctx.mock.value.timestamp({ state: "filled" }),
	color: ({ ctx }) => ctx.mock.value.color({ state: "filled" }),
	number: ({ ctx }) => ctx.mock.value.number({ state: "filled" }),
	keyText: ({ ctx }) => ctx.mock.value.keyText({ state: "filled" }),
	select: ({ ctx }) =>
		ctx.mock.value.select({
			model: ctx.mock.model.select({ options: ["foo", "bar"] }),
			state: "filled",
		}),
	boolean: ({ ctx }) => ctx.mock.value.boolean(),
	geoPoint: ({ ctx }) => ctx.mock.value.geoPoint({ state: "filled" }),
	integration: ({ ctx }) => ctx.mock.value.integration({ state: "filled" }),
	linkToWeb: ({ ctx }) =>
		ctx.mock.value.link({ type: "Web", withTargetBlank: true }),
})

testMigrationImageFieldPatching("patches image fields", {
	regular: ({ migration }) => migration.createAsset("foo", "foo.png"),
	richText: ({ ctx, migration }) => [
		...ctx.mock.value.richText({ pattern: "short", state: "filled" }),
		migration.createAsset("foo", "foo.png"),
	],
})

testMigrationLinkFieldPatching("patches link fields", {
	existingDocument: ({ existingDocuments }) => existingDocuments[0],
	migrationDocument:
		({ migration, migrationDocuments }) =>
		() =>
			migration.getByUID(migrationDocuments[0].type, migrationDocuments[0].uid),
	otherRepositoryContentRelationship: ({ ctx, existingDocuments }) => {
		const contentRelationship = ctx.mock.value.link({ type: "Document" })
		contentRelationship.id = existingDocuments[0].id

		return contentRelationship
	},
})
