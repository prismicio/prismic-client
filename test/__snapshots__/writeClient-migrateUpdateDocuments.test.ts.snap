// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`patches image fields > regular > group 1`] = `
{
  "group": [
    {
      "field": {
        "alt": null,
        "copyright": null,
        "dimensions": {
          "height": 1,
          "width": 1,
        },
        "edit": {
          "background": "transparent",
          "x": 0,
          "y": 0,
          "zoom": 0,
        },
        "id": "ef95d5daa4d",
        "url": "https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=2560&h=1705&fit=crop",
      },
    },
  ],
}
`;

exports[`patches image fields > regular > shared slice 1`] = `
{
  "slices": [
    {
      "id": "2ff58c741ba",
      "items": [
        {
          "field": {
            "alt": null,
            "copyright": null,
            "dimensions": {
              "height": 1,
              "width": 1,
            },
            "edit": {
              "background": "transparent",
              "x": 0,
              "y": 0,
              "zoom": 0,
            },
            "id": "f2c3f8bfc90",
            "url": "https://images.unsplash.com/photo-1604537529428-15bcbeecfe4d?w=4240&h=2832&fit=crop",
          },
        },
      ],
      "primary": {
        "field": {
          "alt": null,
          "copyright": null,
          "dimensions": {
            "height": 1,
            "width": 1,
          },
          "edit": {
            "background": "transparent",
            "x": 0,
            "y": 0,
            "zoom": 0,
          },
          "id": "f2c3f8bfc90",
          "url": "https://images.unsplash.com/photo-1604537529428-15bcbeecfe4d?w=4240&h=2832&fit=crop",
        },
        "group": [
          {
            "field": {
              "alt": null,
              "copyright": null,
              "dimensions": {
                "height": 1,
                "width": 1,
              },
              "edit": {
                "background": "transparent",
                "x": 0,
                "y": 0,
                "zoom": 0,
              },
              "id": "f2c3f8bfc90",
              "url": "https://images.unsplash.com/photo-1604537529428-15bcbeecfe4d?w=4240&h=2832&fit=crop",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "at",
      "variation": "tortor",
      "version": "a79b9dd",
    },
  ],
}
`;

exports[`patches image fields > regular > slice 1`] = `
{
  "slices": [
    {
      "id": "e93cc3e4f28",
      "items": [
        {
          "field": {
            "alt": null,
            "copyright": null,
            "dimensions": {
              "height": 1,
              "width": 1,
            },
            "edit": {
              "background": "transparent",
              "x": 0,
              "y": 0,
              "zoom": 0,
            },
            "id": "45306297c5e",
            "url": "https://images.unsplash.com/photo-1604537466608-109fa2f16c3b?w=4240&h=2832&fit=crop",
          },
        },
      ],
      "primary": {
        "field": {
          "alt": null,
          "copyright": null,
          "dimensions": {
            "height": 1,
            "width": 1,
          },
          "edit": {
            "background": "transparent",
            "x": 0,
            "y": 0,
            "zoom": 0,
          },
          "id": "45306297c5e",
          "url": "https://images.unsplash.com/photo-1604537466608-109fa2f16c3b?w=4240&h=2832&fit=crop",
        },
      },
      "slice_label": "Pellentesque",
      "slice_type": "nulla_posuere",
    },
  ],
}
`;

exports[`patches image fields > regular > static zone 1`] = `
{
  "field": {
    "alt": null,
    "copyright": null,
    "dimensions": {
      "height": 1,
      "width": 1,
    },
    "edit": {
      "background": "transparent",
      "x": 0,
      "y": 0,
      "zoom": 0,
    },
    "id": "c5c95f8d3ac",
    "url": "https://images.unsplash.com/photo-1604537466608-109fa2f16c3b?w=4240&h=2832&fit=crop",
  },
}
`;

exports[`patches image fields > richText > group 1`] = `
{
  "group": [
    {
      "field": [
        {
          "spans": [],
          "text": "Nulla Aliquet Porttitor Lacus Luctus",
          "type": "heading2",
        },
        {
          "alt": null,
          "copyright": null,
          "dimensions": {
            "height": 1,
            "width": 1,
          },
          "edit": {
            "background": "transparent",
            "x": 0,
            "y": 0,
            "zoom": 0,
          },
          "id": "cdfdd322ca9",
          "type": "image",
          "url": "https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=6000&h=4000&fit=crop",
        },
      ],
    },
  ],
}
`;

exports[`patches image fields > richText > shared slice 1`] = `
{
  "slices": [
    {
      "id": "2ff58c741ba",
      "items": [
        {
          "field": [
            {
              "spans": [],
              "text": "Eu Mi",
              "type": "heading3",
            },
            {
              "alt": null,
              "copyright": null,
              "dimensions": {
                "height": 1,
                "width": 1,
              },
              "edit": {
                "background": "transparent",
                "x": 0,
                "y": 0,
                "zoom": 0,
              },
              "id": "bc31787853a",
              "type": "image",
              "url": "https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=4554&h=3036&fit=crop",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [],
            "text": "Fringilla Phasellus Faucibus Scelerisque Eleifend Donec Pretium",
            "type": "heading3",
          },
          {
            "alt": null,
            "copyright": null,
            "dimensions": {
              "height": 1,
              "width": 1,
            },
            "edit": {
              "background": "transparent",
              "x": 0,
              "y": 0,
              "zoom": 0,
            },
            "id": "bc31787853a",
            "type": "image",
            "url": "https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=4554&h=3036&fit=crop",
          },
        ],
        "group": [
          {
            "field": [
              {
                "spans": [],
                "text": "Arcu",
                "type": "heading1",
              },
              {
                "alt": null,
                "copyright": null,
                "dimensions": {
                  "height": 1,
                  "width": 1,
                },
                "edit": {
                  "background": "transparent",
                  "x": 0,
                  "y": 0,
                  "zoom": 0,
                },
                "id": "bc31787853a",
                "type": "image",
                "url": "https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=4554&h=3036&fit=crop",
              },
            ],
          },
        ],
      },
      "slice_label": null,
      "slice_type": "at",
      "variation": "tortor",
      "version": "a79b9dd",
    },
  ],
}
`;

exports[`patches image fields > richText > slice 1`] = `
{
  "slices": [
    {
      "id": "e93cc3e4f28",
      "items": [
        {
          "field": [
            {
              "spans": [],
              "text": "Habitasse platea dictumst quisque sagittis purus sit",
              "type": "o-list-item",
            },
            {
              "alt": null,
              "copyright": null,
              "dimensions": {
                "height": 1,
                "width": 1,
              },
              "edit": {
                "background": "transparent",
                "x": 0,
                "y": 0,
                "zoom": 0,
              },
              "id": "7963dc361cc",
              "type": "image",
              "url": "https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=4554&h=3036&fit=crop",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [],
            "text": "Ac Orci Phasellus Egestas Tellus",
            "type": "heading5",
          },
          {
            "alt": null,
            "copyright": null,
            "dimensions": {
              "height": 1,
              "width": 1,
            },
            "edit": {
              "background": "transparent",
              "x": 0,
              "y": 0,
              "zoom": 0,
            },
            "id": "7963dc361cc",
            "type": "image",
            "url": "https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=4554&h=3036&fit=crop",
          },
        ],
      },
      "slice_label": "Pellentesque",
      "slice_type": "nulla_posuere",
    },
  ],
}
`;

exports[`patches image fields > richText > static zone 1`] = `
{
  "field": [
    {
      "spans": [],
      "text": "Elementum Integer",
      "type": "heading6",
    },
    {
      "alt": null,
      "copyright": null,
      "dimensions": {
        "height": 1,
        "width": 1,
      },
      "edit": {
        "background": "transparent",
        "x": 0,
        "y": 0,
        "zoom": 0,
      },
      "id": "0bbad670dad",
      "type": "image",
      "url": "https://images.unsplash.com/photo-1470071459604-3b5ec3a7fe05?w=7372&h=4392&fit=crop",
    },
  ],
}
`;

exports[`patches link fields > existingDocument > group 1`] = `
{
  "group": [
    {
      "field": {
        "id": "058058c",
        "isBroken": false,
        "lang": "laoreet",
        "link_type": "Document",
        "tags": [],
        "type": "orci",
      },
    },
  ],
}
`;

exports[`patches link fields > existingDocument > shared slice 1`] = `
{
  "slices": [
    {
      "id": "2ff58c741ba",
      "items": [
        {
          "field": {
            "id": "0d137ed",
            "isBroken": false,
            "lang": "tortor,",
            "link_type": "Document",
            "tags": [
              "Risus In",
            ],
            "type": "dui",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "0d137ed",
          "isBroken": false,
          "lang": "tortor,",
          "link_type": "Document",
          "tags": [
            "Risus In",
          ],
          "type": "dui",
        },
        "group": [
          {
            "field": {
              "id": "0d137ed",
              "isBroken": false,
              "lang": "tortor,",
              "link_type": "Document",
              "tags": [
                "Risus In",
              ],
              "type": "dui",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "at",
      "variation": "tortor",
      "version": "a79b9dd",
    },
  ],
}
`;

exports[`patches link fields > existingDocument > slice 1`] = `
{
  "slices": [
    {
      "id": "e93cc3e4f28",
      "items": [
        {
          "field": {
            "id": "382a11c",
            "isBroken": false,
            "lang": "sapien",
            "link_type": "Document",
            "tags": [
              "Aenean",
            ],
            "type": "amet",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "382a11c",
          "isBroken": false,
          "lang": "sapien",
          "link_type": "Document",
          "tags": [
            "Aenean",
          ],
          "type": "amet",
        },
      },
      "slice_label": "Pellentesque",
      "slice_type": "nulla_posuere",
    },
  ],
}
`;

exports[`patches link fields > existingDocument > static zone 1`] = `
{
  "field": {
    "id": "edeaf3a",
    "isBroken": false,
    "lang": "eget",
    "link_type": "Document",
    "tags": [],
    "type": "phasellus",
  },
}
`;

exports[`patches link fields > migrationDocument > group 1`] = `
{
  "group": [
    {
      "field": {
        "id": "9f3857a",
        "isBroken": false,
        "lang": "a",
        "link_type": "Document",
        "tags": [
          "Odio Eu",
        ],
        "type": "amet",
        "uid": "Non sodales neque",
      },
    },
  ],
}
`;

exports[`patches link fields > migrationDocument > shared slice 1`] = `
{
  "slices": [
    {
      "id": "2ff58c741ba",
      "items": [
        {
          "field": {
            "id": "aa66b08",
            "isBroken": false,
            "lang": "gravida",
            "link_type": "Document",
            "tags": [],
            "type": "leo",
            "uid": "Blandit volutpat maecenas",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "aa66b08",
          "isBroken": false,
          "lang": "gravida",
          "link_type": "Document",
          "tags": [],
          "type": "leo",
          "uid": "Blandit volutpat maecenas",
        },
        "group": [
          {
            "field": {
              "id": "aa66b08",
              "isBroken": false,
              "lang": "gravida",
              "link_type": "Document",
              "tags": [],
              "type": "leo",
              "uid": "Blandit volutpat maecenas",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "at",
      "variation": "tortor",
      "version": "a79b9dd",
    },
  ],
}
`;

exports[`patches link fields > migrationDocument > slice 1`] = `
{
  "slices": [
    {
      "id": "e93cc3e4f28",
      "items": [
        {
          "field": {
            "id": "dfcf6a9",
            "isBroken": false,
            "lang": "ultrices",
            "link_type": "Document",
            "tags": [
              "Ipsum Consequat",
            ],
            "type": "eget",
            "uid": "Amet dictum sit",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "dfcf6a9",
          "isBroken": false,
          "lang": "ultrices",
          "link_type": "Document",
          "tags": [
            "Ipsum Consequat",
          ],
          "type": "eget",
          "uid": "Amet dictum sit",
        },
      },
      "slice_label": "Pellentesque",
      "slice_type": "nulla_posuere",
    },
  ],
}
`;

exports[`patches link fields > migrationDocument > static zone 1`] = `
{
  "field": {
    "id": "3f9fce9",
    "isBroken": false,
    "lang": "nisi,",
    "link_type": "Document",
    "tags": [],
    "type": "eros",
    "uid": "Scelerisque mauris pellentesque",
  },
}
`;

exports[`patches link fields > otherRepositoryContentRelationship > group 1`] = `
{
  "group": [
    {
      "field": {
        "id": "058058c",
        "isBroken": false,
        "lang": "nulla",
        "link_type": "Document",
        "tags": [
          "Convallis Posuere",
        ],
        "type": "vitae",
        "url": "https://ac.example",
      },
    },
  ],
}
`;

exports[`patches link fields > otherRepositoryContentRelationship > shared slice 1`] = `
{
  "slices": [
    {
      "id": "2ff58c741ba",
      "items": [
        {
          "field": {
            "id": "0d137ed",
            "isBroken": false,
            "lang": "faucibus",
            "link_type": "Document",
            "tags": [
              "Ultrices",
            ],
            "type": "egestas_integer",
            "url": "https://scelerisque.example",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "0d137ed",
          "isBroken": false,
          "lang": "fringilla",
          "link_type": "Document",
          "tags": [],
          "type": "amet",
          "url": "https://nec.example",
        },
        "group": [
          {
            "field": {
              "id": "0d137ed",
              "isBroken": false,
              "lang": "neque",
              "link_type": "Document",
              "tags": [],
              "type": "lacus",
              "url": "https://cras.example",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "at",
      "variation": "tortor",
      "version": "a79b9dd",
    },
  ],
}
`;

exports[`patches link fields > otherRepositoryContentRelationship > slice 1`] = `
{
  "slices": [
    {
      "id": "e93cc3e4f28",
      "items": [
        {
          "field": {
            "id": "382a11c",
            "isBroken": false,
            "lang": "enim,",
            "link_type": "Document",
            "tags": [],
            "type": "felis_donec",
            "url": "https://sem.example",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "382a11c",
          "isBroken": false,
          "lang": "ac",
          "link_type": "Document",
          "tags": [],
          "type": "enim_praesent",
          "url": "https://sit.example",
        },
      },
      "slice_label": "Pellentesque",
      "slice_type": "nulla_posuere",
    },
  ],
}
`;

exports[`patches link fields > otherRepositoryContentRelationship > static zone 1`] = `
{
  "field": {
    "id": "edeaf3a",
    "isBroken": false,
    "lang": "elementum",
    "link_type": "Document",
    "tags": [],
    "type": "aenean",
    "url": "https://volutpat.example",
  },
}
`;
